### ISSUE TITLE
EcalTPG_updateWeightGroup_test and EcalTPG_updateWeightIdMap_test fail occasionally

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#38964]
The EcalTPG_updateWeightGroup_test failed in CMSSW_12_5_X_2022-08-03-1100 with
```
----- Begin Fatal Exception 03-Aug-2022 12:31:59 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalTPGWeightGroupAnalyzer/'Test1'
Exception Message:
Could not insert a new row in the table ( CORAL : "ITableDataEditor::insertRow" from "CORAL/RelationalPlugins/sqlite" ) from PoolDBOutputService::writeMany 
----- End Fatal Exception -------------------------------------------------
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc10/CMSSW_12_5_X_2022-08-03-1100/unitTestLogs/CondTools/Ecal#/

and EcalTPG_updateWeightIdMap_test with
```
----- Begin Fatal Exception 03-Aug-2022 12:31:59 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalTPGWeightIdMapAnalyzer/'Test1'
Exception Message:
database is locked ( CORAL : "SQLiteStatement::execute" from "CORAL/RelationalPlugins/sqlite" ) from PoolDBOutputService::writeMany 
----- End Fatal Exception -------------------------------------------------
```
in el8_amd64_gcc10, but succeeded in slc7_amd64_gcc10

**Comment by makortel** [comment]
Tests were added in https://github.com/cms-sw/cmssw/pull/38856 that was merged in CMSSW_12_5_X_2022-08-02-2300

**Comment by makortel** [comment]
assign db

**Comment by makortel** [comment]
In CMSSW_12_5_X_2022-08-02-2300 EcalTPG_updateWeightGroup_test failed in ROOT6 IB with
```
----- Begin Fatal Exception 03-Aug-2022 03:31:06 CEST-----------------------
An exception of category 'StdException' occurred while
   [0] Calling endJob for module ExTestEcalTPGWeightGroupAnalyzer/'Test1'
Exception Message:
A std::exception was thrown.
m_stmt is NULL (SQLiteStatement::prepare failed or m_stmt has already been deleted?) ( CORAL : "SQLiteStatement::fetchNext" from "CORAL/RelationalPlugins/sqlite" )
----- End Fatal Exception -------------------------------------------------
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc10/CMSSW_12_5_ROOT6_X_2022-08-02-2300/unitTestLogs/CondTools/Ecal#/

and both tests failed in DEVEL IB
```
----- Begin Fatal Exception 03-Aug-2022 00:31:16 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalTPGWeightGroupAnalyzer/'Test1'
Exception Message:
database is locked ( CORAL : "SQLiteStatement::execute" from "CORAL/RelationalPlugins/sqlite" ) from PoolDBOutputService::forceInit 
----- End Fatal Exception -------------------------------------------------
return code is 65

---> test EcalTPG_updateWeightGroup_test had ERRORS

----- Begin Fatal Exception 03-Aug-2022 00:31:16 CEST-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalTPGWeightIdMapAnalyzer/'Test1'
Exception Message:
database is locked ( CORAL : "SQLiteStatement::execute" from "CORAL/RelationalPlugins/sqlite" ) from PoolDBOutputService::forceInit 
----- End Fatal Exception -------------------------------------------------
return code is 65

---> test EcalTPG_updateWeightIdMap_test had ERRORS
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc10/CMSSW_12_5_DEVEL_X_2022-08-02-2300/unitTestLogs/CondTools/Ecal#/

**Comment by tvami** [comment]
@valsdav can you please have a look?

**Comment by valsdav** [comment]
Hi @tvami sorry for the late reply. I'm investigating with experts. I will let you know.

**Comment by tvami** [comment]
@valsdav do you have any news?

**Comment by valsdav** [comment]
Hi @tvami, I have checked our scripts and they are writing on local sqlite files, having no problems when running on my lxplus CMSSW instance. Can the issue be related to the filesystem in the testing machine? May I ask your help in understanding this issues?

The only clue I have is that two tests were writing on the same sqlite .db file, maybe if they are run in parallel that can create problems with the sqlite library? I can fix this easily in a new PR.

**Comment by tvami** [comment]
Yes, I think the issue really is that the sqlite files have the same name and are in the same directory.

**Comment by ggovi** [comment]
Hello, providing two different sqlite file names for the two tests should fix the issue.

**Comment by valsdav** [comment]
Ok, sorry for the pain, probably testing locally one thing at a time I missed the issue before.. Preparing the small PR to fix this.

**Comment by tvami** [comment]
+db
 * resolved in https://github.com/cms-sw/cmssw/pull/39235
